### ISSUE TITLE
Automaticaly detect cstack global variable

### ISSUE THREAD

**Issue by cmorin6** [nneonneo/ghidra-wasm-plugin#3]
It would be nice if the Pre-Analyzer could automatically detect the global variable used to store the cstack pointer.

There could be a default cstack Pre-Analyzer option (`-2` for exemple) to specify that we want to automatic detection.

A simple option would be to check for known global names in the .name sections (see #2). For example emscripten seem to use "__stack_pointer" but this would only work for limited used case.

An other option would be to let `WasmFunctionAnalysis` try detecting known stack initialization patterns similarly to [this](https://github.com/cmorin6/ghidra-wasm-plugin/blob/master/src/main/java/wasm/analysis/flow/WasmFunctionFlowAnalysis.java#L198). The first `WasmFunctionAnalysis` that would detect a known initialization pattern would then override the default parameter passed to successive `WasmFunctionAnalysis.applyContext()` to avoid duplicating the detection.

**Comment by nneonneo** [comment]
Checking for known global names is only going to work in a very small subset of situations, so I prefer to avoid that. Analysis is a better option: there’s usually just one stack pointer, and the code patterns are quite predictable so it should be easy to pick up the right global. 

I suspect that we can simplify the heuristic: for each function, if it starts with a global.get, guess that global as the stack pointer. We could then guess the stack pointer by e.g. taking the global that appears most often, which should work fine for Wasm modules with enough functions.

**Comment by cmorin6** [comment]
Agreed, a global name heuristic would not probably not be sufficiently reliable to be of any use.

We can indeed simplify the heuristic since we don't need to know which local is used to store the current stack pointer (contrary to my original solution).

The detection of the global could be done as part in the `WasmFunctionAnalysis.analyzeOpcode()` by storing the global index if the instruction is a `global.get` and theinstruction address matches the function start address.
The `WasmPreAnalyzer` main loop could then be split into two: 
* one loop to compute the analysis for each function and keep tracks of the most used global
* a second one to apply the function analysis with the resulting global and trigger the disassembly of the function

**Comment by nneonneo** [comment]
OK, I made a PR for that - feel free to take a peek and see if it makes sense. I did some minimal testing with samples I had laying around and it picked the right global each time, so it should be a good improvement over not guessing at all :)
